### ISSUE TITLE
Subscriptions not working as expected

### ISSUE THREAD

**Issue by dvlprx** [JasperFx/marten#3167]
Hi marten community,

I'm new to the subscription model, but to me this looks like there is something wrong going on :/

Let's have a simple subscription:

 ```
public class MyMartenSubscription : SubscriptionBase
 {
     public async override Task<IChangeListener> ProcessEventsAsync(
         EventRange page,
         ISubscriptionController controller,
         IDocumentOperations operations,
         CancellationToken cancellationToken)
     {
         if (page.Events.Any())
         {
              Debug.WriteLine(...);
         }

         return NullChangeListener.Instance;
     }
 }
```

and a registration, that should start the subscription for all events, that will happen in the future, but not for the already existing ones:

```
opts.Events.Subscribe(new MyMartenSubscription(), opt =>
{
    opt.Options.SubscribeFromPresent();
});
```

then I get the following error:

```
Marten.IDocumentStore: Error: Marten encountered an exception executing 


Marten.Exceptions.ProgressionProgressOutOfOrderException: Progression 'Identity: MyMartenSubscription:All' is out of order. This may happen when multiple processes try to process the projection
   at Marten.Events.Daemon.Progress.UpdateProjectionProgress.PostprocessAsync(DbDataReader reader, IList`1 exceptions, CancellationToken token)
   at Marten.Internal.Sessions.OperationPage.ApplyCallbacksAsync(DbDataReader reader, IList`1 exceptions, CancellationToken token)
   at Marten.Internal.Sessions.TransactionalConnection.ExecuteBatchPagesAsync(IReadOnlyList`1 pages, List`1 exceptions, CancellationToken token)
   at Marten.Internal.Sessions.TransactionalConnection.ExecuteBatchPagesAsync(IReadOnlyList`1 pages, List`1 exceptions, CancellationToken token)
Marten encountered an exception executing 

Marten.Events.Daemon.Internals.SubscriptionAgent: Error: Error trying to process subscription MyMartenSubscription:All

System.NotSupportedException: Specified method is not supported.
   at Marten.Events.Daemon.Internals.ProjectionUpdateBatch.Marten.Services.ISessionWorkTracker.get_Streams()
   at Marten.Events.EventGraph.TryCreateTombstoneBatch(DocumentSessionBase session, UpdateBatch& batch)
   at Marten.Internal.Sessions.DocumentSessionBase.tryApplyTombstoneEventsAsync(CancellationToken token)
   at Marten.Internal.Sessions.DocumentSessionBase.ExecuteBatchAsync(IUpdateBatch batch, CancellationToken token)
   at Marten.Subscriptions.SubscriptionExecution.executeRange(EventRange range)
   at Marten.Subscriptions.SubscriptionExecution.executeRange(EventRange range)
   at Marten.Subscriptions.SubscriptionExecution.executeRange(EventRange range)
Error trying to process subscription MyMartenSubscription:All
```

QUESTION: Is there any way, to get more helpful information about what is going wrong? "Specified method is not supported" doesn't help me a lot and the code it points to is a one liner, doing just a check and throwing the ProgressionProgressOutOfOrderException.

So just to go on, let's remove the SubscribeFromPresent and now register to 1 specific event type:

```
opts.Events.Subscribe(new MyMartenSubscription(), opt =>
{
    // register to 1 specific eventType
    opt.IncludeType<SomeCoolEventType>();
});
```

Now we don't get any errors, but the ProcessEventsAsync method is called with page.Events being empty for all events, that we have not included into the whitelist. Why are we called for events we have not subscribed to?

Any help or pointing in the right direction would be very appreciated, so I can try to figure out for myself how to use that mechanism.

**Comment by jeremydmiller** [comment]
If you're getting `ProgressionProgressOutOfOrderException`, it's very likely that you're somehow running multiple copies of the async daemon simultaneously.

**Comment by jeremydmiller** [comment]
And I can't think of any reason why you would ever be hitting the tombstone event workflow from the subscription. You weren't trying to append *other* events inside the subscription were you? That's not supported yet. 

Sorry, but the behavior you're reporting makes zero sense, and I'm assuming there's something else that's going on.

**Comment by jeremydmiller** [comment]
Definitely some issues with the way the subscriptions are dealing with the error workflow this issue is bringing out, but I think I'm going to have to ask for a little more detail on how you're making this come out. The "progression out of order exception" might be from Marten not setting the progression correctly at the start up point -- but that's happily passing in its test coverage right now.

Can you do a reproduction for this by chance? And I can harden Marten regardless so you don't get the extraneous noise about the tombstone events that shouldn't even be running here.

**Comment by jeremydmiller** [comment]
@dvlprx I wrote a test that reproduced the error. And uncovered something related to configuration that might explain your filtering. I also permanently "fixed" the exception handling of batch failures in a subscription so it doesn't even try to do the tombstone process that's invalid in this context anyway. 

Fixes in Marten 7.10 later today, depending on how hard it is to get another feature in

**Comment by dvlprx** [comment]
Nice .. thanks a lot, marten is really a pleasure to work with - I use it in combination with project orleans.

Just to be complete: could you give a short answer to the second question? Is this also resolved now, or is it by definition, that a subscription is called with page.Events being empty?
